src/com/google/javascript/jscomp/ConvertIIFEArgsToVars.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.javascript.jscomp;
+
+import com.google.common.base.Preconditions;
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Convert the arguments of an immediately invoked function expression to variables
+ * and assign the value from the call arguments. This improves function inlining.
+ *
+ * Since passes which implement OptimizeCalls only handle named references, this is it's own pass.
+ *
+ * @author chadkillingsworth@gmail.com (Chad Killingsworth)
+ */
+public class ConvertIIFEArgsToVars implements CompilerPass, NodeTraversal.Callback {
+  private final AbstractCompiler compiler;
+
+  public ConvertIIFEArgsToVars(AbstractCompiler compiler) {
+    this.compiler = compiler;
+  }
+
+  @Override
+  public void process(Node externs, Node root) {
+   NodeTraversal.traverseEs6(compiler, root, this);
+  }
+
+  public boolean shouldTraverse(NodeTraversal t, Node n, Node parent) {
+    return true;
+  }
+
+  @Override
+  public void visit(NodeTraversal t, Node n, Node parent) {
+    if (!n.isFunction() || parent == null) {
+      return;
+    }
+
+    Node call;
+
+    // Check for standard IIFE
+    // (function() { })()
+    if (parent.isCall() && parent.getFirstChild() == n) {
+      call = parent;
+
+      // Check for explicit calls
+      // (function() {} ).call(null)
+    } else if (parent.isGetProp() && parent.getFirstChild() == n &&
+        n.getNext() != null && n.getNext().isString() &&
+        n.getNext().getString().equals("call") &&
+        parent.getParent() != null && parent.getParent().isCall() &&
+        parent.getParent().getFirstChild() == parent) {
+      call = parent.getParent();
+    } else {
+      return;
+    }
+
+    Node funcParams = n.getSecondChild();
+    if (!funcParams.hasChildren() || NodeUtil.isVarArgsFunction(n)) {
+      return;
+    }
+
+    Node param = funcParams.getFirstChild();
+    List<Node> newVars = new ArrayList<>();
+    while (param != null) {
+      newVars.add(IR.var(param.cloneNode(), getCallArgument(param, call))
+          .useSourceInfoFromForTree(param));
+
+      param = param.getNext();
+    }
+    funcParams.removeChildren();
+    for (int i = newVars.size() - 1; i >= 0; i--) {
+      funcParams.getNext().addChildToFront(newVars.get(i));
+    }
+
```

Comment:
Add comment here:

// Remove all arguments of the call node, except "this" if it exists.